frontend/packages/noobaa-storage-plugin/src/components/bucket-class/wizard-pages/backingstore-page.tsx

```diff
@@ -0,0 +1,255 @@
+import * as React from 'react';
+import * as classNames from 'classnames';
+import * as _ from 'lodash';
+import * as fuzzy from 'fuzzysearch';
+import { FormGroup, TextInput, Title, Form } from '@patternfly/react-core';
+import {
+  Table,
+  TableVariant,
+  TableGridBreakpoint,
+  TableHeader,
+  TableBody,
+  IRowData,
+} from '@patternfly/react-table';
+import { K8sResourceKind, referenceForModel } from '@console/internal/module/k8s';
+import { ResourceLink, Firehose, FirehoseResult } from '@console/internal/components/utils';
+import { NooBaaBackingStoreModel } from '@console/noobaa-storage-plugin/src/models';
+import { connectToModel } from '@console/internal/kinds';
+import { Action, State, BackingStoreStateType } from '../state';
+import { HelpTextCreator } from './help-text';
+
+const tableColumnClasses = [
+  classNames('col-md-4', 'col-sm-4', 'col-xs-6', 'nb-bc-bs-table__data'),
+  classNames('col-md-3', 'col-sm-4', 'col-xs-5', 'nb-bc-bs-table__data'),
+  classNames('col-md-2', 'col-sm-3', 'hidden-xs'),
+  classNames('col-md-2', 'hidden-sm', 'hidden-xs'),
+];
+
+const nameMap = Object.freeze({
+  'aws-s3': 'awsS3',
+});
+
+const columns = [
+  {
+    title: 'Name',
+    props: { className: tableColumnClasses[0] },
+  },
+  {
+    title: 'BucketName',
+    props: { className: tableColumnClasses[1] },
+  },
+  {
+    title: 'Type',
+    props: { className: tableColumnClasses[2] },
+  },
+  {
+    title: 'Region',
+    props: { className: tableColumnClasses[3] },
+  },
+];
+
+const filterSelected = (list: BackingStoreStateType[], tableId: number) => {
+  const sort = (x, y) => x.id.localeCompare(y.id);
+  return list.filter((e) => e.selectedBy === tableId || e.selectedBy === '').sort(sort);
+};
+
+const getTableRows = (list: K8sResourceKind[]) => {
+  return list.reduce((acc, bs) => {
+    const obj = {
+      cells: [],
+      selected: false,
+      id: bs.metadata.name,
+      selectedBy: '',
+    };
+    const type: string = nameMap[_.get(bs, ['spec', 'type'])];
+    obj.cells = [
```

Comment:
refer this comment https://github.com/openshift/console/pull/3007#discussion_r340069737

Reply:
ack